diesel_cli/src/print_schema.rs

```diff
@@ -594,6 +632,64 @@ impl Display for TableDefinitions<'_> {
     }
 }
 
+/// Calculates groups of tables that are related by foreign key.
+///
+/// Given the graph of all tables and their foreign key relations, this returns the set of connected
+/// components of that graph.
+fn foreign_key_table_groups<'a>(
+    tables: &'a [TableData],
+    fk_constraints: &'a [ForeignKeyConstraint],
+) -> Vec<Vec<&'a TableName>> {
+    let mut visited = BTreeSet::new();
+    let mut components = vec![];
+
+    // Find connected components in table graph. For the intended purpose of this function, we treat
+    // the foreign key relation as being symmetrical, i.e. we are operating on the undirected graph.
+    //
+    // The algorithm is not optimized and suffers from repeated lookups in the foreign key list, but
+    // it should be sufficient for typical table counts from a few dozen up to a few hundred tables.
```

Comment:
I feel like we probably want to sort the output so as to make it easy to preserve it later if we want to change the algorithm.